packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationModule.m

```diff
@@ -0,0 +1,225 @@
+// Copyright 2019-present 650 Industries. All rights reserved.
+
+#import <UMCore/UMAppLifecycleService.h>
+#import <UMCore/UMDefines.h>
+#import <UMCore/UMEventEmitterService.h>
+#import <UMCore/UMModuleRegistryProvider.h>
+
+#import <EXScreenOrientation/EXScreenOrientationModule.h>
+#import <EXScreenOrientation/EXScreenOrientationUtilities.h>
+#import <EXScreenOrientation/EXScreenOrientationRegistry.h>
+
+#import <UIKit/UIKit.h>
+
+@interface EXScreenOrientationModule ()
+
+@property (nonatomic, weak) EXScreenOrientationRegistry *screenOrienationRegistry;
+@property (nonatomic, weak) id<UMEventEmitterService> eventEmitter;
+@property (nonatomic, assign) bool hasListeners;
+@property (nonatomic, assign) UIInterfaceOrientation currentScreenOrientation;
+
+@end
+
+@implementation EXScreenOrientationModule
+
+UM_EXPORT_MODULE(ExpoScreenOrientation);
+
+- (void)dealloc
+{
+  [self stopObserving];
+  
+  dispatch_sync(dispatch_get_main_queue(), ^{
+    [[UIDevice currentDevice] endGeneratingDeviceOrientationNotifications];
+  });
+}
+
+- (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
+{
+  _eventEmitter = [moduleRegistry getModuleImplementingProtocol:@protocol(UMEventEmitterService)];
+  _screenOrienationRegistry = [moduleRegistry getSingletonModuleForName:@"EXScreenOrientationRegistry"];
+  id<UMAppLifecycleService> lifercycleService = [moduleRegistry getModuleImplementingProtocol:@protocol(UMAppLifecycleService)];
+
+  if (lifercycleService) {
+    [lifercycleService registerAppLifecycleListener:self];
+  }
+  
+  _hasListeners = NO;
+  
+  dispatch_async(dispatch_get_main_queue(), ^{
+    if (![[UIDevice currentDevice] isGeneratingDeviceOrientationNotifications]){
+      [[UIDevice currentDevice] beginGeneratingDeviceOrientationNotifications];
+    }
+  });
+  
+  UIInterfaceOrientationMask currentOrientationMask = [[[[UIApplication sharedApplication] keyWindow] rootViewController] supportedInterfaceOrientations];
+  [_screenOrienationRegistry moduleDidForeground:self];
+  [_screenOrienationRegistry setMask:currentOrientationMask forModule:self];
```

Comment:
Let's move this call before `moduleDidForeground` so the change goes straight to the desired mask.